tech/languages/go/go-programs.md

```diff
@@ -16,6 +16,13 @@ $ cd $HOME/go/src/hello
 $ touch hello.go
 ```
 
+If you get errors about missing packages, you may need to set a $GOROOT environment variable also. Fedora's `golang-src` package installs common packages into a GOROOT at `/usr/lib/golang` which should clear up most missing packages. Some distributions use `/usr/lib/golang` so this may be confusing for Fedora users reading docs from other distros. If you don't have root or sudo access, you may need to download your own [local GOROOT](https://go.dev/doc/manage-install) archive with basic packages into your home dir.
+
+```console
+$ sudo dnf install golang-src
+$ export GOROOT=/usr/lib/golang
+```
```

Comment:
As I said above if you follow the install that is a pre-requirement to this page, you would not need this.

IIRC golang is a statically compiled language. Therefore installing Golang libraries or Golang runtime of any kind is not needed for a built and distributed binary and is useless in an environment without a compiler. Am I correct in this assumption? If so, then what is your use-case of install just the golang-src?

Reply:
In this case external github imports weren't working because of failed dependencies but I had a custom GOROOT set which wasn't right. Fedora default seems fine if golang-src is installed otherwise non-root users can download an archive of go for a local GOROOT.